src/pages/BlockDetails.vue

```diff
@@ -0,0 +1,339 @@
+      container-class="transactions-list-page__container"
+      @next-page="table.nextPage()"
+      @prev-page="table.prevPage()"
+      @set-page="table.setPage($event)"
+      @set-size="table.setSize($event)"
+    >
+      <template #row="{ item }: { item: Transaction }">
+        <div class="transactions-list-page__row">
+          <TransactionStatus
+            type="tooltip"
+            class="transactions-list-page__icon"
+            :committed="item.committed"
+          />
+
+          <div class="transactions-list-page__column">
+            <div class="transactions-list-page__label">TxID</div>
+
+            <BaseHash
+              :hash="item.hash"
+              :type="hashType"
+              :link="'/transactions/' + item.hash"
+              copy
+            />
+
+            <div class="transactions-list-page__time">
+              {{ format(item.payload.creation_time) }}
+            </div>
+          </div>
+
+          <div class="transactions-list-page__column">
+            <div class="transactions-list-page__label">Block</div>
+
+            <BaseLink :to="'/blocks/' + item.block_height">
+              {{ item.block_height }}
+            </BaseLink>
+          </div>
+        </div>
+      </template>
+    </BaseTable>
+  </BaseContentBlock>
+</template>
+
+<script setup lang="ts">
+import { ref, onBeforeMount } from 'vue';
+import { useRoute } from 'vue-router';
+import { useBlockTable } from '~shared/lib/table';
+import { useWindowSize, computedEager } from '@vueuse/core';
+import { http } from '~shared/api';
+import { TransactionStatus } from '~entities/transaction';
+import { format } from '~shared/lib/time';
+import {
+  filterTransactionsModel as ftm,
+  TransactionStatusFilter,
+  TransactionTypeFilter,
+} from '~features/filter-transactions';
+
+import BaseContentBlock from '~base/BaseContentBlock.vue';
+import BaseTable from '~base/BaseTable.vue';
+import BaseHash from '~shared/ui/components/BaseHash.vue';
+import BaseLink from '~shared/ui/components/BaseLink.vue';
+import ArrowIcon from '~icons/arrow.svg';
+
+const HASH_BREAKPOINT = 1200;
+
+const status = ref<ftm.Status>(null);
+const tab = ref<ftm.Tab>('all');
+
+const { width } = useWindowSize();
+
+const hashType = computedEager(() =>
+  width.value < HASH_BREAKPOINT ? 'short' : 'full',
+);
+
+let table = useBlockTable([]);
+const block = ref();
+
+onBeforeMount(() => {
+  // Access the route object and get the 'id' parameter
+  const route = useRoute(); // Use useRoute to access the $route object
+  const heightString = Array.isArray(route.params.id)
+    ? route.params.id.join('')
+    : route.params.id;
+  const height = parseInt(heightString, 10); // Convert the string to an integer
+  console.log(height);
+  const fetchData = async () => {
+    try {
+      const blockDetails = await http.fetchBlocksDetails(height);
+      block.value = blockDetails;
+      console.log(blockDetails);
+      table = useBlockTable(blockDetails.transactions);
```

Comment:
You should avoid using composables outside of the scope of the component.

`useBlockTable` will initialize reactive logic on invocation (such as refs, reactive objects, watchers, hooks etc). In order for this reactive logic to work properly (e.g. destroy watchers when the component is disposed), it should be initialised within component's _setup_ stage, or during its lifecycle hooks.

In this case, however, you invoke an async function at any time and then initialize reactive data within `useBlockTable`. Vue will not track the reactivity properly.